### PR TITLE
PR review reminders

### PR DESCRIPTION
**Pull request by melinath**
Note: this is a large PR but most of it is tests. This will run as a dry run for a while so I can double-check the output and make sure everything's fine before implementing & moving to non-dry-run.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

File: .ci/magician/cmd/scheduled_pr_reminders.go
Comment (shuyama1) [inline .ci/magician/cmd/scheduled_pr_reminders.go:311]: I think the overall system looks good to me. There's only one specific scenario that I'm not certain about - comments after approval from the same contributor. I wonder if we should consider sending reminders to reviewers instead of contributors in such cases, as if the following comment should block the PR merge, the reviewer should probably dismiss the previous approval instead. One common reason that I can think of for this case is reviewer leave comment to announce pause on merging PRs due to probably CI issues.

File: .github/workflows/scheduled-pr-reminders.yml
Comment (shuyama1) [inline .github/workflows/scheduled-pr-reminders.yml:7]: Out of curiosity, why is this particular time chosen?

File: .ci/magician/cmd/scheduled_pr_reminders_test.go
Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders_test.go:224]: "Comment review but no approval" is correct because the "APPROVED" review is overwritten by the "COMMENTED" review (because they have the same reviewer).

For comments that don't signify a request for contributor action, reviewers can leave a normal comment instead of a review of type "COMMENTED". Reviews are the only mechanism we have to signal that there is expected action from contributors; in general, I would expect a "COMMENTED" review to mean that something is blocking the PR besides a specific change (for example, needing more information about the use case). For cases where we don't want to notify the contributor, we can apply the `disable-automatic-closure` label.

File: .ci/magician/cmd/scheduled_pr_reminders_test.go
Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders_test.go:327]: yes

Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders_test.go:327]: specifically, a new review of type "COMMENTED" would dismiss a previous review of type "APPROVED" from the same contributor. New comments that are _not_ part of a review would have no impact.

File: .ci/magician/cmd/scheduled_pr_reminders.go
Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders.go:311]: A new review of type "COMMENTED" would dismiss a previous review of type "APPROVED" from the same contributor. New comments that are not part of a review would have no impact.

I testing this on https://github.com/GoogleCloudPlatform/magic-modules/pull/10495 and GitHub's approval behavior is:

1. I approve. State: approved.
2. I add a COMMENTED review. State: approved.
3. I add a CHANGES_REQUESTED review. State: Changes requested.
4. I dismiss the CHANGES_REQUESTED review. State: Unreviewed.

This end state has 3 reviews which are APPROVED, COMMENTED, and DISMISSED: https://api.github.com/repos/GoogleCloudPlatform/magic-modules/pulls/10495/reviews

This behavior seems clearly inconsistent to me; the dismissed review should be ignored and return the PR to its approved state.

After thinking on it some more, I am ambivalent about this; making COMMENTED not override APPROVED seems like a reasonable compromise.

File: .github/workflows/scheduled-pr-reminders.yml
Comment (melinath) [reply .github/workflows/scheduled-pr-reminders.yml:7]: The time is in UTC; after 9am should mean that the action always runs on the same day in PDT as UTC (in case that could end up being confusing) while still running early in the day for folks in the UK. `37` is arbitrary just to avoid running on the hour in case that's crowded.

File: .ci/magician/cmd/scheduled_pr_reminders.go
Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders.go]: Changed in a86c375ac

File: .ci/magician/cmd/scheduled_pr_reminders_test.go
Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders_test.go:327]: Changed this behavior in a86c375ac - see the other thread about it.

File: .ci/magician/cmd/scheduled_pr_reminders_test.go
Comment (melinath) [reply .ci/magician/cmd/scheduled_pr_reminders_test.go]: changed this behavior in a86c375ac - see the other thread about it.

File: .ci/magician/cmd/scheduled_pr_reminders_test.go
Comment (shuyama1) [inline .ci/magician/cmd/scheduled_pr_reminders_test.go:13]: ~are we running these unit tests anywhere?~

nvm, looks like the tests are included in membership-checker unit tests. The GHA is running all magician unit tests.

File: .github/workflows/scheduled-pr-reminders.yml
Comment (shuyama1) [reply .github/workflows/scheduled-pr-reminders.yml:7]: yea, makes sense. Thanks

**Review by shuyama1 — APPROVED** [review]

**Pull request merged**
